src/core/Track.cpp

```diff
@@ -889,76 +897,85 @@ void TrackContentObjectView::mouseMoveEvent( QMouseEvent * me )
 	const float ppt = m_trackView->trackContainerView()->pixelsPerTact();
 	if( m_action == Move )
 	{
-		const int x = mapToParent( me->pos() ).x() - m_initialMousePos.x();
-		MidiTime t = qMax( 0, (int)
-			m_trackView->trackContainerView()->currentPosition()+
-				static_cast<int>( x * MidiTime::ticksPerTact() /
-									ppt ) );
-		if( ! ( me->modifiers() & Qt::ControlModifier )
-		   && me->button() == Qt::NoButton )
-		{
-			t = t.toNearestTact();
-		}
-		m_tco->movePosition( t );
+		MidiTime newPos = draggedTCOPos( me );
+
+		newPos = max( 0, newPos.getTicks() ); // Don't go left of bar zero
```

Comment:
Your call if you think I need to change it, otherwise I'll leave it as-is.
```suggestion
		// Don't go left of bar zero
		newPos = max( 0, newPos.getTicks() );
```